### PR TITLE
Add rate vs pileup plotting

### PR DESCRIPTION
**Pull request by bundocka**
#### What does this pull request implement/fix? Explain your changes.

Adding plotting for pileup dependency plots for all L1 Calo algos:

- New rate_vs_pileup plotter
- HW_Emu_jetMet_rates updated to automatically run hw vs emu pileup dependence plots
- Event reader and cmsl1t binary updated to take a flag doVertex  (default false) from the config if the ntuples have vertex but do not have other reco information

#### Any other comments?
An example would be if you require another pull request to be merged before this one.
If you do, please reference it.

### CODE REVIEW

File: cmsl1t/playground/eventreader.py [hunk 1]
```diff
@@ -31,12 +31,14 @@
 }
 
 
-def get_trees(load_emu_trees, load_reco_trees):
+def get_trees(load_emu_trees, load_reco_trees, load_vertex_trees):
```
Comment (kreczko) [inline cmsl1t/playground/eventreader.py]: If you are changing the behaviour of crucial classes such as eventreader, please update the tests:
https://github.com/cms-l1t-offline/cms-l1t-analysis/blob/master/test/playground/test_eventreader.py#L11

Alternatively, you can set a sensible default for the new parameter.

Comment (bundocka) [reply cmsl1t/playground/eventreader.py]: Added load_vertex and extra test to test_eventreader.py

File: cmsl1t/plotting/rate_vs_pileup.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+from cmsl1t.hist.factory import HistFactory
+import cmsl1t.hist.binning as bn
+from cmsl1t.utils.draw import draw, label_canvas
+from cmsl1t.recalc.resolution import get_resolution_function
+
+from rootpy.context import preserve_current_style
+from rootpy.plotting import Legend
+
+
+class RateVsPileupPlot(BasePlotter):
```
[the remark this answers is not in the document]
Comment (bundocka) [reply cmsl1t/plotting/rate_vs_pileup.py]: A few things didn't seem particularly intuitive and took a while to get my head round but other than that was fine.

File: cmsl1t/analyzers/HW_Emu_jetMet_rates.py [hunk 3]
```diff
@@ -33,6 +34,18 @@ def extractSums(event):
 
     return online
 
+# Eta ranges so we can put |\eta| < val as the legend header on the
+# efficiency plots.
+ETA_RANGES = dict(
```
[the remark this answers is not in the document]
Comment (bundocka) [reply cmsl1t/analyzers/HW_Emu_jetMet_rates.py]: Yes, was copied from jetMet_analyzer.py. I couldn't see anywhere common that would be obvious to move it to but that would of course be fine.

File: cmsl1t/playground/eventreader.py [hunk 4]
```diff
@@ -31,12 +31,14 @@
...
     selected_trees = ['event', 'upgrade']
     if load_emu_trees:
         selected_trees.extend(['emuCaloTowers', 'emuUpgrade'])
     if load_reco_trees:
         selected_trees.extend(['jetReco', 'metFilterReco', 'recoTree'])
+    if load_vertex_trees and not load_reco_trees:
+        selected_trees.extend(['recoTree'])
```
[the remark this answers is not in the document]
Comment (bundocka) [reply cmsl1t/playground/eventreader.py]: nVtx is in the recoTree. Some ntuples will only have "recoTree" and no other reco trees. So just need this option for that case.

**Pull request merged**